### ISSUE TITLE
[Bug] [AI] Lua Error for "Move to any enemy CA"

### ISSUE THREAD

**Issue by knyghtmare** [wesnoth/wesnoth#6829]
### Game and System Information

- **Version:** 1.17.5+dev
- **Downloaded from:** Github
- **Build info:** Battle for Wesnoth v1.17.5+dev (a573e58007e-Modified)
- **OS:** Windows 10 (msys2)

### Description of the bug

Lua Error Message:
```js
20220703 02:25:55 error scripting/lua: ai/lua/ca_move_to_any_enemy.lua:54: attempt to index a nil value (field 'integer index')
stack traceback:
	ai/lua/ca_move_to_any_enemy.lua:54: in function <ai/lua/ca_move_to_any_enemy.lua:54>
	[C]: in function 'table.sort'
	ai/lua/ca_move_to_any_enemy.lua:54: in function <ai/lua/ca_move_to_any_enemy.lua:16>
	(...tail calls...)
```

### Expected behavior

No error message like it was yesterday.

### Additional context

Might be something to 90f1c01f41364413f551201976433f9e07f792fb

**Comment by knyghtmare** [comment]
Update:
Happens if an event of this order is present in the scenario code

```ini
[event]
        name="side_1_turn"
        first_time_only=no

        [filter_condition]
            [lua]
                code= << return wesnoth.current.turn % 4 == 0 >>
            [/lua]
        [/filter_condition]

        {GENERIC_UNIT 1 "Swordsman" 23 21}
        {GENERIC_UNIT 1 "Swordsman" 23 21}
        {GENERIC_UNIT 1 "Swordsman" 23 21}
[/event]
```

**Comment by mattsc** [comment]
Uh, no.  First of all, there is no way that that would cause it.  Second, I actually tried and put this into The Freelands and let the two sides play it out and nothing happened.

Also, nothing in that code says anything about guardians.

**Comment by mattsc** [comment]
Could you check whether 6ed684af7 fixes this for you?  Your event by itself does not cause the problem for me (guardians or not; it's not sufficient to cause the error as far as I can tell, other conditions need to be satisfied as well) so I created my own test case.  Just want to make sure that this fixes it in your case also before closing this.

**Comment by knyghtmare** [comment]
Yes, it's fixed now.

**Comment by mattsc** [comment]
Fixed in 1.16 now also: 6ff3703d
